include/boost/random/sobol.hpp

```diff
@@ -0,0 +1,1120 @@
+/* boost random/sobol.hpp header file
+ *
+ * Copyright Justinas Vygintas Daugmaudis 2010-2017
+ * Distributed under the Boost Software License, Version 1.0. (See
+ * accompanying file LICENSE_1_0.txt or copy at
+ * http://www.boost.org/LICENSE_1_0.txt)
+ */
+
+#ifndef BOOST_RANDOM_SOBOL_HPP
+#define BOOST_RANDOM_SOBOL_HPP
+
+#include <boost/random/detail/gray_coded_qrng_base.hpp>
+
+#include <limits>
+
+#include <boost/cstdint.hpp>
+#include <boost/integer/static_log2.hpp>
+
+#include <boost/multi_array.hpp>
+
+#include <boost/preprocessor/stringize.hpp>
+
+//!\file
+//!Describes the quasi-random number generator class template sobol.
+//!
+//!\b Note: it is especially useful in conjunction with class template uniform_real.
+
+namespace boost {
+namespace random {
+
+/** @cond */
+namespace detail {
+namespace sbl {
+
+// Maximum allowed space dimension
+#define BOOST_RANDOM_SOBOL_MAX_DIMENSION 1111
```

Comment:
Again, any possibility of getting this an order of magnitude higher?

Reply:
This is possible. Joe and Koe provide Sobol primitive polynomials and direction numbers up to dimension 21201.

http://web.maths.unsw.edu.au/~fkuo/sobol/

I trust this source, but anything else in the order of magnitude range will need to be written down, peer reviewed and published in a serious journal. ;-)

Reply:
I think 21201 is a perfectly acceptable max dimension. It would put the capabilities of this Sobol sequence *way* beyond the capabilities of the GSL (which implements up to dimension 40).